tst/com/amazon/ion/benchmark/DataGeneratorTest.java

```diff
@@ -0,0 +1,209 @@
+                if (BigInteger.TEN.pow(digitCount - 1).compareTo(coefficient) > 0) {
+                    digitCount = digitCount - 1;
+                }
+                assertTrue(digitCount >= range.get(0) && digitCount <= range.get(1));
+            }
+        }
+    }
+
+    /**
+     * Assert the format of generated file is conform with the expected format [ion_binary|ion_text].
+     * @throws Exception if error occurs when executing Ion data generator.
+     */
+    @Test
+    public void testGeneratedFormat() throws Exception {
+        List<String> inputs = new ArrayList<>(Arrays.asList("ion_text","ion_binary"));
+        for (int i = 0; i < 2; i++ ) {
+            Map<String, Object> optionsMap = Main.parseArguments("generate", "-S", "500", "-T", "float", "-f", inputs.get(i), "test4.ion");
+            GeneratorOptions.executeGenerator(optionsMap);
+            String format = ((List<String>)optionsMap.get("--format")).get(0);
+            outputFile = optionsMap.get("<output_file>").toString();
+            Path path = Paths.get(outputFile);
+            byte[] buffer = Files.readAllBytes(path);
+            assertEquals(Format.valueOf(format.toUpperCase()) == Format.ION_BINARY, IonStreamUtils.isIonBinary(buffer));
+        }
+    }
+
+    /**
+     * Assert the unicode code point range of the character which constructed the generated Ion string is conform with the expect range.
+     * @throws Exception if error occurs when executing Ion data generator.
+     */
+    @Test
+    public void testGeneratedStringUniCodeRange() throws Exception {
+        Map<String, Object> optionsMap = Main.parseArguments("generate", "-S", "500", "-T", "string", "-N", "[96,99]","test5.10n");
+        try (IonReader reader = DataGeneratorTest.constructReader(optionsMap)) {
+            List<Integer> range = WriteRandomIonValues.parseRange(optionsMap.get("--text-code-point-range").toString());
+            while (reader.next() != null) {
+                String str = reader.stringValue();
+                for (int i = 0; i < str.length(); i++) {
+                    int codePoint = Character.codePointAt(str, i);
+                    int charCount = Character.charCount(codePoint);
+                    //UTF characters may use more than 1 char to be represented
+                    if (charCount == 2) {
+                        i++;
+                    }
+                    assertTrue(codePoint >= range.get(0) && codePoint <= range.get(1));
+                }
+            }
+        }
+    }
+
+    /**
+     * Assert the generated timestamps is follow the precision and proportion of the given timestamp template.
+     * @throws Exception if error occurs when executing Ion data generator.
+     */
+    @Test
+    public void testGeneratedTimestampTemplateFormat() throws Exception{
+        Map<String, Object> optionsMap = Main.parseArguments("generate", "-S", "500", "-T", "timestamp", "-M", "2010T", "test6.10n");
```

Comment:
Should the timestamp templates be specified like the other ranges (i.e. as an Ion list)?

Reply:
> Should the timestamp templates be specified like the other ranges (i.e. as an Ion list)?

Yes, it should be an Ion list, this part has been updated in the code.